webapp/cube/views.py

```diff
@@ -133,6 +145,135 @@ def cube_microcerts():
     )
 
 
+@login_required
+def get_microcerts():
+
+    # backend_true requires query string,
+    # therefore we need context
+    # also this avoids loading contracts api where not required
+    is_test_backend = (
+        current_app.config["CONTRACTS_TEST_API_URL"]
+        if flask.request.args.get("test_backend", "false")
+        else current_app.config["CONTRACTS_LIVE_API_URL"]
+    )
+    sso_user = user_info(flask.session)
+
+    ua_contracts_api = UAContractsAPI(
+        session=ua_contracts_session,
+        authentication_token=(flask.session.get("authentication_token")),
+        api_url=is_test_backend,
+    )
+
+    edx_url = (
+        f"{edx_api.base_url}/auth/login/tpa-saml/"
+        "?auth_entry=login&idp=ubuntuone&next="
+    )
+
+    edx_user = edx_api.get_user(sso_user["email"]) if sso_user else None
+    product_listings = ua_contracts_api.get_product_listings("canonical-cube")[
+        "productListings"
+    ]
+
+    assertions = {}
+    enrollments = []
+    passed_courses = 0
+    if edx_user:
+        assertions = {
+            assertion["badgeclass"]: assertion
+            for assertion in badgr_api.get_assertions(
+                BADGR_ISSUER, edx_user["email"]
+            )["result"]
+        }
+
+        enrollments = [
+            enrollment["course_details"]["course_id"]
+            for enrollment in edx_api.get_enrollments(edx_user["username"])
+            if enrollment["is_active"]
+        ]
+
+    certified_badge = {}
+    if CERTIFIED_BADGE in assertions:
+        assertion = assertions.pop(CERTIFIED_BADGE)
+        if not assertion["revoked"]:
+            certified_badge["image"] = assertion["image"]
+            certified_badge["share_url"] = assertion["openBadgeId"]
+
+    courses = []
+    for product_list in product_listings:
+        attempts = []
+        course = {
+            "product_listing_id": product_list["id"],
+            "value": product_list["price"]["value"],
+        }
+        edx_ids = [
+            edx_id
+            for edx_id in product_list["externalIDs"]
+            if edx_id["origin"] == "EdX"
+        ][0]
+        for course_id in edx_ids["IDs"]:
```

Comment:
I think we can assume that edx_ids["IDs"] only has one value, so I would just do:
```
course_id = [
    edx_id[0]
    for edx_id in product_list["externalIDs"]
    if edx_id["origin"] == "EdX"
][0]
```
And then you can get rid of the for loop.

Reply:
mm ok. Let's hope they don't change the structure. There is another list inside, so I changed a bit the code.